proposals/null-coalescing-assignment.md

```diff
@@ -37,7 +39,23 @@ assignment_operator
     ;
 ```
 
-Which follows the [existing semantic rules for compound assignment operators](https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#compound-assignment). What that means is that an operation of the form `x ??= y` is processed as if written `x = (x ?? y)` (except `x` shall be computed once). However, we permit the assignment to `x` (of its existing value) to be elided in the case that `x` is not null.
+Which follows the [existing semantic rules for compound assignment operators](https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#compound-assignment), except that we elide the assignment if the left-hand side is non-null. The rules for this feature are as follows.
+
+Given `a ??= b`, where `A` is the type of `a`, `B` is the type of `b`, and `A0` is the underlying value type of `A` if `A` is the nullable type.
```

Comment:
>type [](start = 139, length = 4)

... otherwise the type of A.

Reply:
We don't appear to need A0 in the specification.

---
In reply to: [203555762](https://github.com/dotnet/csharplang/pull/1721#discussion_r203555762) [](ancestors = 203555762)